### PR TITLE
567 Parse docket entry description for cabc

### LINKED ISSUE #566: Error parsing recap email: ValueError: Field 'entry_number' expected a number but got 'No'.
Sentry Issue: [COURTLISTENER-36J](https://sentry.io/organizations/freelawproject/issues/3647360444/?referrer=github_integration)

```
ValueError: invalid literal for int() with base 10: 'No'
  File "django/db/models/fields/__init__.py", line 1823, in get_prep_value
    return int(value)

ValueError: Field 'entry_number' expected a number but got 'No'.
(12 additional frame(s) were not displayed)
...
  File "django/db/models/sql/query.py", line 1370, in build_filter
    condition = self.build_lookup(lookups, col, value)
  File "django/db/models/sql/query.py", line 1216, in build_lookup
    lookup = lookup_class(lhs, rhs)
  File "django/db/models/lookups.py", line 25, in __init__
    self.rhs = self.get_prep_lookup()
  File "django/db/models/lookups.py", line 77, in get_prep_lookup
    return self.lhs.output_field.get_prep_value(self.rhs)
  File "django/db/models/fields/__init__.py", line 1825, in get_prep_value
    raise e.__class__(
```

### PR DESCRIPTION
**Pull request by albertisfu**
Seems that the problem parsing the docket entry descriptions was only for `cabc` NEFs.

So added support to the docket entry description for this court and also add raise an exception when we can't get the description so we can fix it.

This also fixes  #566 the problem here is that the notification doesn't contain a link to the attachment instead it says: `No document attached ` so now we avoid getting the document number in those cases.

### CODE REVIEW

File: juriscraper/pacer/email.py [hunk 1]
```diff
@@ -153,8 +153,11 @@ def _get_document_number(self) -> str:
         :returns: Document number, cleaned
         """
         path = self._sibling_path("Document Number")
-        text = self.tree.xpath(path)[0].text_content()
-        words = re.split(r"\(|\s", clean_string(text))
+        node = self.tree.xpath(path)[0].text_content()
+        text_number = clean_string(node)
+        if text_number == "No document attached":
+            return None
+        words = re.split(r"\(|\s", clean_string(node))
```
Comment (mlissner) [inline juriscraper/pacer/email.py:160]:
```suggestion
        words = re.split(r"\(|\s", text_number)
```

File: tests/examples/pacer/nef/s3/cacb_1.txt [hunk 2]
```diff
@@ -0,0 +1,109 @@
+Return-Path: <cmecfhelpdesk@cacb.uscourts.gov>
+Received: from icmecf201.gtwy.uscourts.gov (icmecf201.gtwy.uscourts.gov [63.241.40.204])
+ by inbound-smtp.us-west-2.amazonaws.com with SMTP id rom4mdiassf2i4p25mm0m39fh9lt9sf1e1asdu01
+ for benheston@recap.email;
```
Comment (mlissner) [inline tests/examples/pacer/nef/s3/cacb_1.txt:4]: Remember that we want to anonymize these?

File: juriscraper/pacer/email.py [hunk 3]
```diff
@@ -222,22 +225,38 @@ def _get_description(self):
...
+                if description:
+                    return description
+
+            node = self.tree.xpath(f"{path}text()")
+            if len(node):
+                for des_part in node:
+                    description = description + des_part
+                description = clean_string(description)
+                if description:
+                    return description
```
Comment (mlissner) [inline juriscraper/pacer/email.py:255]: With this kind of repetitive code, I usually like to do a little loop:

```
for path in possible_paths:
    node = self.tree.xpath(path)
    ...
```
Makes it more clear that the same code is being run over and over. I think you could even just put the appellate logic into the same loop without much trouble too, but I haven't checked that carefully.

**Review by mlissner — CHANGES_REQUESTED** [review]
Just a couple comments.

File: juriscraper/pacer/email.py (hunk 1 above)
Comment (albertisfu) [reply juriscraper/pacer/email.py]: Changed.

File: tests/examples/pacer/nef/s3/cacb_1.txt (hunk 2 above)
Comment (albertisfu) [reply tests/examples/pacer/nef/s3/cacb_1.txt]: Yeah, I've already anonymized also the @recap.email addresses for these and previous email examples.

**Review by mlissner — APPROVED** [review]
Looks good, thanks. Merging.

**Pull request merged**